### ISSUE TITLE
support JSON-schema-like validation

### ISSUE THREAD

**Issue by WebFreak001** [SdlangInitiative/sdlschema#1]
Supporting JSON schema would make it easier to have formats supported in both JSON and SDL.

I have implemented this in typescript for code-d, it might be worth starting off with this kind of code, though it's by no means perfect: https://github.com/Pure-D/code-d/tree/master/src/sdl

**Comment by BradleyChatha** [comment]
I'll have a look into it

**Comment by BradleyChatha** [comment]
I've written up something here: https://github.com/BradleyChatha/jsonschema

It's written in a way where I should be able to just whip up a quick `SdlangAdapter` and things should *just work* (tm), but that's a tomorrow job.

**Comment by BradleyChatha** [comment]
I've added SDLang support into jsonschema, however I'm uncertain how to allow it to handle certain things such as SDLang's attributes, perhaps with non-standard extensions?

**Comment by WebFreak001** [comment]
yes I think that would work, you can see I just added `attributes` additionally to `values` (members) in code-d:

https://github.com/Pure-D/code-d/blob/79b8218c1b02ed253985a5282cc2e6c164530bb9/src/sdl/sdl-contributions.ts#L315
